app/controllers/verify/finance_controller.rb

```diff
@@ -7,13 +7,18 @@ class FinanceController < ApplicationController
     before_action :confirm_step_allowed
     before_action :submit_idv_form, only: [:create]
     before_action :submit_idv_job, only: [:create]
+    before_action :refresh_if_not_ready, only: [:show]
```

Comment:
How about a new controller for each step that polls for the vendor result, and then proceeds with processing the result? So, after the SessionsController submits the data to the vendor, it would redirect to this new Job Polling controller, and when the result is ready, it will execute what we used to have in the `create` action, such as:
```ruby
result = step.submit
analytics.track_event(Analytics::IDV_BASIC_INFO_SUBMITTED_VENDOR, result.to_h)

if result.success?
  process_success
else
  process_failure
end
```
That way, we clean up the controller by removing `before_action`s. I'm thinking SessionsController#create will now look something like this:
```ruby
def create
  initialize_idv_session

  result = idv_form.submit(profile_params)
  analytics.track_event(Analytics::IDV_BASIC_INFO_SUBMITTED_FORM, result.to_h)

  if result.success?
    submit_idv_job
    redirect_to verify_session_job_polling_path
  else
    process_failure
  end
end
```
which is a lot more readable to me than having to look at `before_action`s to figure out what is going on.

Thoughts?